### PR TITLE
THE FUTURE OF MEDICAL IS HERE - INTERGALACTIC HEALTH VOUCHERS

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Every other department (besides sec) has a progression mechanic, usually gated behind RNG, metaknowledge, and materials in order to acquire. Medical doctors meanwhile are pretty much useless after setting up cryo and have very few ways to personally enrich their job experience. So while cargo rides by on ATV's with shotguns slung on their backs, botany has invented the most deadly bannana in existence, robotics is rolling out in Durands with modules that would make a mauler blush, RND and genetics are pretty much demigods... medical doctors at best, sit around healing the other departments so they can keep doing cool shit while the medical doctors get a pat on the head before slinking back to stare at the catgirl nurses.

THAT CHANGES TODAY!

YOU GET POINTS FOR DOING SHIT AND PUSHING YOUR MEDICAL AUTISM TO THE LIMIT AFTER SO MANY CHANGES IM DONE REVISING THE DESCRIPTION. GET POINTS FOR EVERY POINT OF DAMAGE HEALED BY MOBS IN MEDBAY!

This PR has been locally tested and is 100% ready to ACTUALLY MERGE YOU PUSSIES

:cl: Kor
add: Added special rewards to the medical vendor. You can earn points by healing people in medbay. You lose points if people die or are hurt in medbay.
/:cl:

### CODE REVIEW

**Comment by chanoc1** [comment]
By are hurt in medbay do you mean take damage in medbay or they walk in hurt? I'm also assuming the reward from healing them has is more then the loss from damage.

**Comment by KorPhaeron** [comment]
@chanoc1 only if they actually take damage in there

**Comment by chanoc1** [comment]
oh, so fucking up surgery is a no no, neat

what happens if someone storms into medbay and starts murderboning? i feel like that wouldnt be too fair for the doctors but again, maybe the points gained are more than points lost

**Comment by KorPhaeron** [comment]
They will make their points back if they heal the people, and having your department attacked tends to result in lost resources

**Comment by Ausops** [comment]
((((((medbay)))))-sponsored rage cages in the bar soon

**Comment by chanoc1** [comment]
coolio this pr gets an :ok_hand: from me

**Comment by Kromgar** [comment]
This is a joke right? HEALING STAFF for 2k credits?

**Review by lzimann — CHANGES_REQUESTED** [review]
the rewards are a bit subpar

**Comment by KorPhaeron** [comment]
@Kromgar 

I've tested extensively, it takes nearly an hour to get enough points in an average round

**Comment by dannno** [comment]
kor is fat and ugly

**Comment by ma44** [comment]
I can say that these costs are balanced and approved by ma44's seal of "not exploitable"

**Review by ChangelingRain — CHANGES_REQUESTED** [review]
Taking damage in medbay makes you lose points!
If you have a clone pod in main medbay, you'll get a huge point boost every time you clone somebody!

**Comment by ChangelingRain** [comment]
You can farm points with genetics monkeymen and sleepers!

**Comment by ChangelingRain** [comment]
This should probably include basic medical supplies so you can buy more healing items with the points you got from healing people!

**Comment by KorPhaeron** [comment]
@ChangelingRain 

I'll add a client check so you can't abuse monkeymen

**Comment by chanoc1** [comment]
why is the medical equipment vendor still have the mining vendor description? 😕  maybe im just reading the code wrong

File: code/game/machinery/medicalvendor.dm [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/obj/machinery/medical_equipment_vendor
+	name = "mining equipment vendor"
```
Comment (Poojawa) [inline code/game/machinery/medicalvendor.dm]: > Mining equipment.

**Comment by swindly** [comment]
Does the amount of points change when a mob is healed at full health or damaged at 200 damage?

**Comment by Repukan** [comment]
Make the staff more expensive.

Is it the infinite charge staff or the limited charge staffs?

How many charges does it get, if limited?

**Comment by PKPenguin321** [comment]
is this real

File: code/game/machinery/medicalvendor.dm [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	icon = 'icons/obj/machines/mining_machines.dmi'
+	icon_state = "mining"
+	density = 1
+	anchored = 1
+	var/gbp = 0
+	var/list/prize_list = list( //if you add something to this, please, for the love of god, use tabs and not spaces.
+		new /datum/data/mining_equipment("Medical Hardsuit",	/obj/item/clothing/suit/space/hardsuit/ert/med ,			1000),
+		new /datum/data/mining_equipment("Syndicate Medical Cyborg",			/obj/item/weapon/antag_spawner/nuke_ops/borg_tele/medical ,100),
+		new /datum/data/mining_equipment("Medical Beamgun",				/obj/item/weapon/gun/medbeam ,							1500),
+		new /datum/data/mining_equipment("Healing Staff",				/obj/item/weapon/gun/magic/staff/healing ,						2000),
```
Comment (PKPenguin321) [inline code/game/machinery/medicalvendor.dm]: lmao

**Comment by ShadowDeath6** [comment]
>staff of OP infinite full heals for 2k points
>op item so expensive that it's pretty much gated behind war only (with syndicate laws) for 100 points

:+1: :

**Comment by bobdobbington** [comment]
I'd swap it out with a wand of rez for ~1.5k to be honest. Still good but at least it doesn't permanently obsolete medbay.

**Comment by Repukan** [comment]
Swap it with an enhanced medbot tbh.

Here's hoping this is a joke PR because if not wew

**Comment by Jazaen** [comment]
I think Kor finally lost it.

**Comment by CosmicScientist** [comment]
I like how everyone is joining in to make this April Fool's feature appear real.

File: code/game/machinery/medicalvendor.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/game/machinery/medicalvendor.dm]: Forgot to change

**Comment by KorPhaeron** [comment]
@Niknakflak 

Do not downvote my PRs without leaving feedback that is rude

@ShadowDeath6 

I forgot to up the price will fix

@bobdobbington 

Good idea, I'll bump the staff to something ridiculous like 20k

**Comment by Niknakflak** [comment]
Not everything needs a dumb ticket system

**Comment by KorPhaeron** [comment]
Ticket systems are a staple of modern games, people clearly enjoy them, and I think we should listen to our players

**Comment by Niknakflak** [comment]
close this meme PR

**Comment by Ausops** [comment]
If all goes according to the developmental milestones, we'll do a test merge of this revolutionary feature somewhere around early april, maybe for just a day.

**Comment by davidchan13** [comment]
Medbay is gonna be stealing people's shoes, piling glass shards out the door and rubber banding assistants back to the sleepers for heals. I love it :+1:

**Comment by RandomMarine** [comment]
An age where MDs entice grayshirts with promises of nitrile gloves and circular saws in exchange for toolboxing random people in the halls so they can get more points.

This feels perfect in the wacky world of SS13.

**Comment by jughu** [comment]
>syndicate medborg 
Whats the lore of this?

**Comment by FantasticFwoosh** [comment]
POOOOOOOOOOOOOOOOOOOINTS

**Comment by Doppeltongue** [comment]
I can't imagine that anything could possibly go wrong with this, nor will it ever be exploited in hilarious ways. Let's merge it immediately.

**Comment by KingPhilipIII** [comment]
>Syndie mediborg 
>Equipped with emag
>Emag cargo
>Abuse

**Comment by GunHog** [comment]
That very bad feeling when the Design Lead finally snaps. The day the weight of his position and putting up with all the drama, spam, and opposition breaks his back.

You guys did it. You went too far. Now his mind is broken and he is making meme PRs.
### YOU BROKE KOR.

**Comment by Aloraydrel** [comment]
Stupid kor slime is broken

**Comment by GunHog** [comment]
Or the more likely scenario is that Kor is playing early April Fool's jokes.

**Comment by Qbopper** [comment]
if this isn't an april fools joke ill probably end my life

this pr is my fault

File: code/game/machinery/medicalvendor.dm [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	var/dat
+	dat +="<div class='statusDisplay'>"
+	dat += "You have [gbp] GBP collected."
+	dat += "</div>"
+	dat += "<br><b>Rewards point cost list:</b><BR><table border='0' width='300'>"
+	for(var/datum/data/mining_equipment/prize in prize_list)
+		dat += "<tr><td>[prize.equipment_name]</td><td>[prize.cost]</td><td><A href='?src=\ref[src];purchase=\ref[prize]'>Purchase</A></td></tr>"
+	dat += "</table>"
+
+	var/datum/browser/popup = new(user, "miningvendor", "Mining Equipment Vendor", 400, 350)
```
Comment (coiax) [inline code/game/machinery/medicalvendor.dm]: copy paste

**Comment by Bawhoppen** [comment]
great PR :1st_place_medal:x:100:

**Comment by coiax** [comment]
@Robustin and @KorPhaeron are the same. WAKE UP SHEEPLE.

**Comment by FantasticFwoosh** [comment]
@Qbopper this wouldn't be the first time, remember sniper rifle syndicate base ID's and stationwide sniper rifle ID's with a HOP giving out syndicate rifles?

**Comment by ExcessiveUseOfCobblestone** [comment]
FUCK Planetstation

**Comment by RandomMarine** [comment]
Okay, real talk.

We have all these point systems coming in.

We've had PRs for engineering points.
We've had mining points for a long time.
We've had cargo points since basically forever.
Now we're looking into medical points.

What if these systems could be unified?
WHAT IF WE CAN FINALLY HAVE MEANINGFUL CURRENCY THAT'S VALUED BY ALL THE CREW?

**Comment by coiax** [comment]
@RandomMarine I think that's called an economy

**Comment by Jazaen** [comment]
We already have one. It's called "Valids".

**Comment by Cyberboss** [comment]
reeeeeeee

File: code/modules/mob/living/damage_procs.dm [hunk 4]
```diff
@@ -140,6 +140,12 @@
 	bruteloss = Clamp((bruteloss + (amount * config.damage_multiplier)), 0, maxHealth*2)
 	if(updating_health)
 		updatehealth()
+
+	var/area/A = get_area(src)
+	if(istype(A, /area/medical))
+		for(var/obj/machinery/medical_equipment_vendor/V in machines)
```
Comment (octareenroon91) [inline code/modules/mob/living/damage_procs.dm]: YES

File: code/modules/mob/living/damage_procs.dm [hunk 5]
```diff
@@ -151,6 +157,12 @@
 	oxyloss = Clamp((oxyloss + (amount * config.damage_multiplier)), 0, maxHealth*2)
 	if(updating_health)
 		updatehealth()
+
+	var/area/A = get_area(src)
+	if(istype(A, /area/medical))
+		for(var/obj/machinery/medical_equipment_vendor/V in machines)
+			V.gbp += amount
```
Comment (octareenroon91) [inline code/modules/mob/living/damage_procs.dm]: also copy paste

**Comment by FantasticFwoosh** [comment]
Its disheartening that people actually want this PR.

**Comment by Bawhoppen** [comment]
its not like its related to the time of the year or anything

**Comment by FantasticFwoosh** [comment]
FFS not even irish.

**Comment by Bawhoppen** [comment]
I AIN'T TALKING ABOUT ST. PADDY'S DAY

**Comment by optimumtact** [comment]
closed for the same reasons as the engineering points' pr

**Comment by davidchan13** [comment]
FNR

**Comment by CosmicScientist** [comment]
RIP fun.

**Comment by adamkad1** [comment]
Oranges bad

**Comment by Fikou** [comment]
tru

**Comment by ExcessiveUseOfCobblestone** [comment]
imagine having to unsub from something over 2 years ago
